.github/workflows/push-pull.yml

```diff
@@ -0,0 +1,20 @@
+name: Push-Pull
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  pyiron:
+    uses: pyiron/actions/.github/workflows/push-pull.yml@pyiron_base_lessons
+    secrets: inherit
+    with:
+      do-commit-updated-env: false
+      do-build-docs: false  # the build-docs action tries to use sphinx, not jupyterbook
+      unit-test-dir: tests
+      coveralls-and-codacy-test-dir: tests
```

Comment:
Can we select `coveralls` and `codacy` independently?

Reply:
Yeah, sure. Nice idea.

I just pushed a commit to the branch we're currently targeting that slaps feature flags on these steps so they can be disabled independently: https://github.com/pyiron/actions/pull/113/commits/feeecf66caa8c91b7cb66a3a107109c2608e75f9

And then broke the current `do-coveralls-and-codacy` flag down to its constituent parts in the main `push-pull` workflow: https://github.com/pyiron/actions/pull/113/commits/823fee013dcd0c68734b6d890ff403d5ae31088b